debug/org.eclipse.debug.tests/src/org/eclipse/debug/tests/ui/LaunchConfigurationTabGroupViewerTest.java

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 SAP SE and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     SAP SE - initial version
+ *******************************************************************************/
+
+package org.eclipse.debug.tests.ui;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfigurationType;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationPresentationManager;
+import org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationsDialog;
+import org.eclipse.debug.ui.IDebugUIConstants;
+import org.eclipse.debug.ui.ILaunchConfigurationDialog;
+import org.eclipse.debug.ui.ILaunchConfigurationTab;
+import org.eclipse.debug.ui.ILaunchConfigurationTabGroup;
+import org.eclipse.swt.widgets.Display;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LaunchConfigurationTabGroupViewerTest {
+
+	private static interface ThrowingRunnable<T extends Exception> {
+		void run() throws T;
+	}
+
+	private static final String LAUNCH_CONFIG_TYPE_ID = "org.eclipse.debug.ui.tests.launchConfigurationType1";
+	private static final String LAUNCH_CONFIG_MODE = ILaunchManager.RUN_MODE;
+	private ILaunchConfigurationType fLaunchConfigurationType;
+	private LaunchConfigurationsDialog fLaunchConfigurationsDialog;
+
+	@Before
+	public void createDialog() throws CoreException {
+		fLaunchConfigurationType = getLaunchManager().getLaunchConfigurationType(LAUNCH_CONFIG_TYPE_ID);
+		ILaunchConfigurationTabGroup tabGroup = getLaunchConfigurationTabGroup(fLaunchConfigurationType);
+
+		fLaunchConfigurationsDialog = (LaunchConfigurationsDialog) createLaunchConfigurationDialog();
+		tabGroup.createTabs(fLaunchConfigurationsDialog, ILaunchManager.RUN_MODE);
```

Comment:
I added some assertions to make sure there are at least 2 tabs, only `SpyTab`s and only from different subclasses (using the same type might not work since `org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.setActiveTab(Class<? extends ILaunchConfigurationTab>)` returns the 1st tab of a kind which would break a test under special circumstances